src/compiler/core.ts

```diff
@@ -1250,7 +1250,7 @@ namespace ts {
         for (let { value: pair, done } = iterator.next(); !done; { value: pair, done } = iterator.next()) {
             const [key, value] = pair;
             const result = callback(value, key as (string & __String));
-            if (result) {
+            if (result !== undefined && result !== false as any) {

```

Comment:
the forEach* functions are meant to check for falsy values in general.. where is the location that misuses the function?

Reply:
`markerName` in `fourslash.ts`.